### PR TITLE
Rework dev to use static deployment with nginx proxy

### LINKED ISSUE #1522: Switch testpilot.dev.mozaws.net to static deployment from master
We auto-deploy `master` branch via Docker to an Elastic Beanstalk instance at `testpilot.dev.mozaws.net`

We auto-deploy `development-static` branch via `gulp` &`aws sync` to an s3 bucket distributed through CloudFront at `testpilot-static.dev.mozaws.net`

But, we no longer need these to be separate.

We should: 
1. Switch `master` branch to static deployment to the S3 bucket `testpilot-static.dev.mozaws.net` 
   - (or create a new S3 bucket named `testpilot.dev.mozaws.net`)
2. Switch `testpilot.dev.mozaws.net` to point at the CloudFront host behind `testpilot-static.dev.mozaws.net` 
   - (or create a new CloudFront distro for the new S3 bucket)
3. Switch to using [`bin/deploy.sh`](https://github.com/mozilla/testpilot/blob/master/bin/deploy.sh) rather than [just bare `aws sync` in `circle.yml`](https://github.com/mozilla/testpilot/blob/master/circle.yml#L75)
4. Retire Docker-based build and deployment from our `circle.yml` setup 

Either way, all the above is stuff I / we set up and self-service on Cloud Services AWS. So, this doesn't require bothering ops for help.

### PR DESCRIPTION
**Pull request by lmorchard**
- Update add-on dev environment to use HTTPS
- Remove {stage,dev}-static envs from add-on
- Master branch uses bin/deploy.sh to upload to S3 bucket
- Upload **version** to S3 (only dev)
- Add Dockerized nginx header filter proxy
- Enable caching for node_modules in circle.yml
- Re-enable setup-test-dependencies.sh to download Firefox binary for
  add-on testing, but disable integration test package install within
  the script (folloup to PR #1624)
- Further bin/circleci cleanup

Fixes #1522, Issue #1509

### CODE REVIEW

**Comment by lmorchard** [comment]
I tested out this PR by using `1522-master-static` in place of `master` in `circle.yml` to auto-update https://testpilot.dev.mozaws.net/ on pushes to my own branch.

Also did some things outside of this PR, infrastructure-wise:
- Created a new S3 bucket named `testpilot.dev.mozaws.net` - _without_ web site hosting enabled (see Issue #1575)
- Created a new Docker container for the header transform proxy and [uploaded it to Docker Hub](https://hub.docker.com/r/lmorchard/txp-proxy/) - this proxies requests for the S3 bucket
- Launched a new EC2 instance named `Test Pilot dev HTTP proxy` (ID: `i-0a49cc4d7ea523a5e`) with Docker installed and started an instance of `lmorchard/txp-proxy`
- Created a CloudFront distribution (`E2ERG47PHCWD0Z`) pointing at the proxy in EC2

The EC2 instance should never need to be updated unless we're also making changes to prod & stage, so there's no auto-updating there. The only thing this PR auto-updates is the S3 bucket.

Not entirely sure where / if to document this stuff. I don't think it's exactly worth keeping secret, since you shouldn't be able to do anything to these resourced without a Cloud Services AWS login

File: addon/index.js [hunk 1]
```diff
@@ -41,29 +41,17 @@ const SERVER_ENVIRONMENTS = {
     BADGE_COLOR: '#AA00AA'
   },
   dev: {
-    BASE_URL: 'http://testpilot.dev.mozaws.net',
+    BASE_URL: 'https://testpilot.dev.mozaws.net',
```
Comment (ghost) [inline addon/index.js:44]: yesss

**Review by ghost — APPROVED** [review]
This looks reasonable to me, and is a pain to test without all the infrastructure set up.  Let's push to -dev!

File: bin/deploy.sh [hunk 2]
```diff
@@ -107,3 +107,12 @@ for fn in $(find dist -name 'index.html' -not -path 'dist/index.html'); do
...
 done
+
+# __version__ JSON; 10 minute cache
+aws s3 cp \
+  --cache-control "max-age=${MAX_AGE}" \
+  --content-type "application/json" \
+  --metadata "{${HPKP}, ${HSTS}, ${TYPE}}" \
+  --metadata-directive "REPLACE" \
+  --acl "public-read" \
+  version.json s3://${TESTPILOT_BUCKET}/__version__
```
Comment (relud) [inline bin/deploy.sh:118]: does putting this here mean that on every deploy we delete `/__version__` at line 89 above, and reupload it here?

Comment (lmorchard) [reply bin/deploy.sh:118]: It shouldn't... `version.json` is generated in the root directory of the project, not `frontend/build`, so it shouldn't get caught up with the sync at line 89. And `__version__` is never actually a local filename

Comment (lmorchard) [reply bin/deploy.sh:118]: Oh wait, derp, I see what you mean... yeah, it gets deleted & re-uploaded because there's never a local `__version__` file.

Comment (relud) [reply bin/deploy.sh:118]: the sync at line 89 deletes any files not in `dist/` (because of the --delete flag on line 90)

Comment (relud) [reply bin/deploy.sh:118]: so `__version__` _should_ be a local filename, to prevent deleting

**Pull request merged**